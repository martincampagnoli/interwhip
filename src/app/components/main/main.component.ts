import { ChangeDetectionStrategy, Component } from '@angular/core';

@Component({
  selector: 'app-main',
  templateUrl: './main.component.html',
  styleUrls: ['./main.component.scss'],
  standalone: true,
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [],
})
export class MainComponent {}
