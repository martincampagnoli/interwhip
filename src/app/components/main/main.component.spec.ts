import { ComponentFixture, TestBed } from '@angular/core/testing';

import { MainComponent } from './main.component';
import { RouterTestingModule } from '@angular/router/testing';

describe('MainComponent', () => {
  let component: MainComponent;
  let fixture: ComponentFixture<MainComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [MainComponent, RouterTestingModule],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(MainComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
