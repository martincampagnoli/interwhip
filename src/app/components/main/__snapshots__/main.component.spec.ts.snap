// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="card-container"
  >
    <app-button-link>
      <a
        class="card w-100 pb-3 pt-3"
        href="/placeholder"
        rel="noopener"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
          data-mat-icon-type="font"
          role="img"
        >
          grid_on
        </mat-icon>
        <span>
          Placeholder button
        </span>
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
          data-mat-icon-type="font"
          role="img"
        >
          chevron_right
        </mat-icon>
      </a>
    </app-button-link>
  </div>
</div>
`;
